libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigationNotification.kt

```diff
@@ -0,0 +1,372 @@
+package com.mapbox.services.android.navigation.v5.navigation
+
+import android.app.Notification
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import android.os.Build
+import android.text.SpannableString
+import android.text.TextUtils
+import android.text.format.DateFormat
+import android.widget.RemoteViews
+import androidx.core.app.NotificationCompat
+import androidx.core.content.ContextCompat
+import com.mapbox.api.directions.v5.DirectionsCriteria.IMPERIAL
+import com.mapbox.api.directions.v5.models.LegStep
+import com.mapbox.navigator.BannerInstruction
+import com.mapbox.services.android.navigation.R
+import com.mapbox.services.android.navigation.v5.navigation.notification.NavigationNotification
+import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress
+import com.mapbox.services.android.navigation.v5.utils.DistanceFormatter
+import com.mapbox.services.android.navigation.v5.utils.LocaleUtils
+import com.mapbox.services.android.navigation.v5.utils.time.TimeFormatter.formatTime
+import java.util.Calendar
+
+/**
+ * This is in charge of creating the persistent navigation session notification and updating it.
+ */
+internal class MapboxNavigationNotification : NavigationNotification {
+
+    private val END_NAVIGATION_ACTION = "com.mapbox.intent.action.END_NAVIGATION"
+    private val SET_BACKGROUND_COLOR = "setBackgroundColor"
```

Comment:
Same here. We've done / followed this pattern as part of other classes conversion.